### DISCUSSION TITLE
zig build static library in windows should define SODIUM_STATIC

### DISCUSSION THREAD

**Discussion by sgwong** [jedisct1/libsodium#1278] [General]
I encounter linking error with zig build in windows when trying to buid another dynamic library which use sodium static library
lld-link: error: duplicate symbol: DllMain

It seems like when build static sodium library in windows it still include the DllMain. After reading the source code, I need to add the following line in the build.zig to resolve the issue.(line 89)
lib.defineCMacro("SODIUM_STATIC", "1");

**Comment by jedisct1** [comment]
That was fixed in d787d2b1cf13ad2e69c3a7ebc3fb7b68b6430774 -- Thanks!
